devito/builtins.py

```diff
@@ -205,6 +205,24 @@ def fset(f, g):
     return f
 
 
+def nbl_to_padsize(nbl, ndim):
```

Comment:
I'm tried, so maybe reading this function wrong, but doesn't look like it currently supports input of the form `((20, 30), 20, 10)` etc. which it probably should i.e. Just need to modify this
```
        if len(as_tuple(nb_t[0])) == 1:
            nbl = tuple((nbd, nbd) for nbd in nb_t)
```
to loop through each dim and do the check.

Reply:
Rushed it indeed nice catch, fixed